src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -662,7 +662,7 @@ private static void waitForAgentOnline(JenkinsRule r, String name, Map<String, L
             throw new AssertionError("no such agent: " + name);
         }
         if (!(node instanceof Slave)) {
-            throw new AssertionError("agent is not a Slave: " + name);
+            throw new AssertionError("agent is not an agent: " + name);
```

Comment:
I am sorry that I missed this in my first review, but the previous message is the more correct message because it is reporting that `node` has the wrong type.  It is expecting an instance of type `Slave`
```suggestion
            throw new AssertionError("agent is not a Slave: " + name);
```